packages/next-urql/src/init-urql-client.ts

```diff
@@ -13,6 +13,9 @@ export function initUrqlClient(clientOptions: ClientOptions): Client | null {
       ...clientOptions,
       suspense: isServer || clientOptions.suspense,
     });
+    // Serialize the urqlClient to null on the client-side.
+    // This ensures we don't share client and server instances of the urqlClient.
+    (urqlClient as any).toJSON = () => null;
```

Comment:
This in general is a really odd hack 😆  But I hope it's working. Is this a critical hotfix?

Reply:
Not critical, more anticipating someone to walk into this pitfall